hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractLeafQueue.java

```diff
@@ -644,8 +644,8 @@ public void validateSubmitApplication(ApplicationId applicationId,
           !(this instanceof AutoCreatedLeafQueue)) {
         String msg =
             "Queue " + getQueuePath() + " already has " + getNumApplications()
-                + " applications,"
-                + " cannot accept submission of application: " + applicationId;
+                + " applications, what is more than the max " + getMaxApplications()
+                + ", so cannot accept submission of application: " + applicationId;
```

Comment:
This message is misleading, because exceeding the max applications is not the only reason why this could be logged and an AccessControlException can be thrown. I don't think we should log this.